src/lib/plot/hint.js

```diff
@@ -52,75 +62,89 @@ class Hint extends PureRenderComponent {
       value: React.PropTypes.object,
       format: React.PropTypes.func,
       orientation: React.PropTypes.oneOf([
-        ORIENTATION_AUTO,
-        ORIENTATION_BOTTOMLEFT,
-        ORIENTATION_BOTTOMRIGHT,
-        ORIENTATION_TOPLEFT,
-        ORIENTATION_TOPRIGHT
+        ORIENTATION.AUTO,
+        ORIENTATION.TOPLEFT,
+        ORIENTATION.BOTTOMLEFT,
+        ORIENTATION.TOPRIGHT,
+        ORIENTATION.BOTTOMRIGHT,
+        ORIENTATION.EDGETOP_LEFT,
+        ORIENTATION.EDGEBOTTOM_LEFT,
+        ORIENTATION.EDGETOP_RIGHT,
+        ORIENTATION.EDGEBOTTOM_RIGHT,
+        ORIENTATION.TOP_EDGELEFT,
+        ORIENTATION.BOTTOM_EDGELEFT,
+        ORIENTATION.TOP_EDGERIGHT,
+        ORIENTATION.BOTTOM_EDGERIGHT
       ])
     };
   }
 
   static get defaultProps() {
     return {
       format: defaultFormat,
-      orientation: ORIENTATION_AUTO
+      orientation: ORIENTATION.AUTO
     };
   }
 
   /**
    * Get the right coordinate of the hint.
+   * When x undefined or null, edge case, pin right.
    * @param {number} x X.
    * @returns {{right: *}} Mixin.
    * @private
    */
   _getCSSRight(x) {
     const {
       innerWidth,
-      marginRight} = this.props;
+      marginRight
+    } = this.props;
     return {
-      right: `${marginRight + innerWidth - x}px`
+      right: x === undefined || x === null ? 0 : marginRight + innerWidth - x
```

Comment:
Nitpick: I would early return and use only `null` rather than allow `undefined`, since allowing both allows ambiguity and `undefined` is ambiguous.

Also, early return because when it's `null`, there's no need for `innerWidth` or `marginRight`. It may improve readability slightly.